### ISSUE TITLE
Question about dice loss used in utils.py

### ISSUE THREAD

**Issue by SamsungApple** [Beckschen/TransUNet#67]
Thank you for your work.
While I was studying your code to implement to my data, I happen to have a problem understanding dice coefficient loss in utils.py file.
The dice loss used is

def _dice_loss(self, score, target):
        target = target.float()
        smooth = 1e-5
        intersect = torch.sum(score * target)
        y_sum = torch.sum(target * target)
        z_sum = torch.sum(score * score)
        loss = (2 * intersect + smooth) / (z_sum + y_sum + smooth)
        loss = 1 - loss
        return loss

But from what I know denominator of the dice coefficient is the sum of each target and input sum,
but the sum of square of each score and target.  
Is there a reason for the calculation?

Thank you.

**Comment by hdnminh** [comment]
Hi @yuyinzhou  @Beckschen @andife 

I hope all you have a great last week of the year!

Can you explain your implemented dice loss:
```
    def _dice_loss(self, score, target):
        target = target.float()
        smooth = 1e-5
        intersect = torch.sum(score * target)
        y_sum = torch.sum(target * target)
        z_sum = torch.sum(score * score)
        loss = (2 * intersect + smooth) / (z_sum + y_sum + smooth)
        loss = 1 - loss
        return loss
```
Why are **y_sum** and **z_sum** calculated by square of target and score tensor, respectively? Following the formula, we just sum it basically without squaring itself.

**Comment by hdnminh** [comment]
Hi @SamsungApple,
Have you figured it out yet? Please share me. Thank you in advance
